packages/opentelemetry-core/test/utils/deep-merge.test.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as assert from 'assert';
+import deepMerge from '../../src/utils/deep-merge';
+
+describe('deepMerge', () => {
+  it('should deep merge two objects', () => {
```

Comment:
I would prefer having more tests here specially with more `undefined` and `null` as values.

Reply:
Thanks, added one: `it('should nullify and undefine as per source props'...`